### ISSUE TITLE
sros_config does not run commands correctly in Ansible 2.3

### ISSUE THREAD

**Issue by jurajama** [ansible/ansible#24874]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

Module sros_config

##### ANSIBLE VERSION

```
ansible 2.3.0.0
  config file =
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, Nov  6 2016, 00:28:07) [GCC 4.8.5 20150623 (Red Hat 4.8.5-11)]
```

##### OS / ENVIRONMENT
CentOS-7

##### SUMMARY

sros_config module is not running commands input from file properly in Ansible 2.3. The very basic functionality of feeding in tab-aligned input files causes failures as Ansible is running commands on wrong levels of CLI.

##### STEPS TO REPRODUCE

When I'm using this input file for sros_config:
```
configure
    system
        name "{{ inventory_hostname }}"
        time
            ntp
                ntp-server authenticate
```

The result is that sros_config tries to run "_ntp-server authenticate_" directly under "_config>system>time_" in SROS, while it should be run in level "config>system>time>*ntp*". Now command "ntp" is not executed at all and therefore "bad command" error appears.
The same input file **worked fine with Ansible 2.2**.

```
"msg": "ntp-server authenticate\r\n                                  ^\r\nError: Bad command. \r\n*A:cont2-dev2>config>system>time# "
```

##### EXPECTED RESULTS

sros_config shall follow the command hierarchy given in input file and run command "ntp" before running "ntp-server authenticate"

This is how the commands are run manually:
```
*A:cont2-dev2# configure
*A:cont2-dev2>config# system
*A:cont2-dev2>config>system# time
*A:cont2-dev2>config>system>time# ntp
*A:cont2-dev2>config>system>time>ntp# ntp-server authenticate
```

##### ACTUAL RESULTS

```
fatal: [cont2-dev2 -> localhost]: FAILED! => {
    "changed": false,
    "command": "ntp-server authenticate",
    "failed": true,
    "invocation": {
        "module_args": {
            "backup": false,
            "config": null,
            "defaults": false,
            "host": null,
            "lines": null,
            "match": "line",
            "parents": null,
            "password": null,
            "port": null,
            "provider": {
                "host": "10.2.210.64",
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
                "port": null,
                "ssh_keyfile": null,
                "timeout": null,
                "transport": "cli",
                "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
            },
            "save": true,
            "src": "configure\n    system\n        name \"cont2-dev2\"\n        time\n            ntp \n                ntp-server authenticate\n",
            "ssh_keyfile": null,
            "timeout": null,
            "username": null
        }
    },
    "msg": "ntp-server authenticate\r\n                                  ^\r\nError: Bad command. \r\n*A:cont2-dev2>config>system>time# ",
    "rc": 1
}
```

**Comment by ansibot** [comment]
cc @privateip @gundalow @Qalthos
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jurajama** [comment]
@privateip : I'm afraid that this fix is not yet fully working. 

I installed devel-branch Ansible on virtualenv, the version is this:
`ansible 2.4.0 (devel 60ce6438e3) last updated 2017/08/14 13:48:25`

Then I tried to apply this configuration to SROS:

```
    system
        name "{{ inventory_hostname }}"
        time
            ntp
                ntp-server authenticate
            zone EET

```
The result is that part "ntp-server authenticate" is now ok, so the fix works for that part. 

However the next command "zone EET" after that fails because sros_config tries to run it in the "ntp" command context like it would be a subcommand of ntp, while actually it is in parallel to ntp in the SROS command hierarchy. The same input works fine with Ansible 2.2.		

```
fatal: [cont1-dev2 -> localhost]: FAILED! => {
    "changed": false,
    "command": "zone EET",
    "failed": true,
    "invocation": {
        "module_args": {
            "backup": false,
            "config": null,
            "defaults": false,
            "host": null,
            "lines": null,
            "match": "line",
            "parents": null,
            "password": null,
            "port": null,
            "provider": {
                "host": "10.2.210.63",
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
                "port": null,
                "ssh_keyfile": null,
                "timeout": null,
                "transport": "cli",
                "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
            },
            "save": true,
            "src": "exit all\nconfigure\n#--------------------------------------------------\necho \"System Configuration\"\n#--------------------------------------------------\n    system\n        name \"cont1-dev2\"\n        time\n            ntp\n                ntp-server authenticate\n            zone EET\n",
            "ssh_keyfile": null,
            "timeout": null,
            "username": null
        }
    },
    "msg": "zone EET\r\n                                      ^\r\nError: Bad command. \r\n*A:cont1-dev2>config>system>time>ntp# ",
    "rc": 1
}
```

It is not only this specific command group where the failure occurs, but there are many other similar places in SROS config hierarchy.

**Comment by jurajama** [comment]
Here's tip for the others who may face the same issue that I reported above still occurring with Ansible 2.4.
I got sros_config working in 2.4 by adding "exit" command in those places of the input file where the command hierarchy returns to previous level like in the following example after configuring "ntp", need to go one level up to configure "zone".

To make the same input file work with both 2.2 and 2.4, the exit commands need to be inside conditionals.

```
    system
        name "{{ inventory_hostname }}"
        time
            ntp
                ntp-server authenticate
{% if ansible_version.full | version_compare('2.4.0.0', '>=') %}
                exit
{% endif %}

            zone EET
```
